kinto/core/storage/postgresql/__init__.py

```diff
@@ -533,9 +533,19 @@ def purge_deleted(self, collection_id, parent_id, before=None,
             placeholders['before'] = before
 
         with self.client.connect() as conn:
-            result = conn.execute(query.format_map(safeholders), placeholders)
+            result = conn.execute(delete_tombstones.format_map(safeholders), placeholders)
+            deleted = result.rowcount
+
+            # If purging everything from a parent_id, then clear timestamps.
+            if collection_id is None and before is None:
```

Comment:
Thinking about this more, it's a little strange that purging tombstones is the thing that triggers removal from the `timestamps` collection. If I delete a bucket, surely its "collection" timestamps should be deleted before I delete any records that are "under it" or purge their tombstones? Maybe the problem is just that `purge_deleted` is a bad name for this method, and we should change the concept to something like `recursive_delete` (which is outside the scope of this PR).